R/new-blog-post.R

```diff
@@ -0,0 +1,82 @@
+#' Create a new blog post
+#'
+#' Creates (and potentially opens) the `index.qmd` file for a new blog post.
+#'
+#' @param title A character string for the title of the post. It is converted
+#' to title case via [tools::toTitleCase()].
+#' @param dest A character string (or NULL) for the path within `posts`. By
+#' default, the title is adapted as the directory name.
+#' @param open A logical: have the default editor open a window to edit the
+#' `index.qmd` file?
+#' @param call A call object for reporting errors.
+#' @param ... A named list of values to be added to the yaml header, such as
+#' `description`, `author`, `categories`, etc.
+#' @return The path to the index file.
+#' @export
+#' @examples
+#' \dontrun{
+#'  \donttest{
+#' new_blog_post("making quarto blog posts", categories = c("R"))
+#'
+#'  }
+#' }
+#'
+new_blog_post <- function(title, dest = NULL, open = rlang::is_interactive(),
+                          call = rlang::current_env(), ...) {
+
+  if (is.null(dest)) {
+    # Scrub title to make directory name
+    dest <- gsub("[[:space:]]", "-", tolower(title))
+  }
+  dest_path <- make_post_dir(dest, call)
+  post_yaml <- make_post_yaml(title, ...)
+  qmd_path <- write_post_yaml(post_yaml, dest_path, call)
+  if (open) {
+    utils::file.edit(qmd_path)
+  }
+  invisible(qmd_path)
+}
+
+make_post_dir <- function(dest, call) {
+  working <- fs::path_wd()
+
+  post_path <- fs::path(working, "posts", dest)
```

Comment:
Should we try to make sure this is called at root of the quarto project ? 

So that if this function is called elsewhere in the project directory we don't create a post folder where we wouldn't want ? 

Maybe this check could be in `new_blog_post()` 

- Checking we are at root of the project
- If no erroring, or tweaking all path to be from project root maybe.

Related to some thinking at #180 (but this one is really about computing path during knitting)